### DISCUSSION TITLE
plot_deflected_shape(speed = Q_(500, "RPM")  ---> error? only "rad/s" possible?

### DISCUSSION THREAD

**Discussion by KimYoungJoong** [petrobras/ross#1136] [Q&A] [answered]
[4_241210_modeling.zip](https://github.com/user-attachments/files/18281784/4_241210_modeling.zip)
Hello everyone! I'm YJ Kim from South Korea.

Currently I'm doing a lateral rotordynamics for a vertical hydro power generator especially for Unbalance Response.

For convenience, I prefer to use "RPM" rather than "rad/s" to interpret  the plot to my colleagues so I used "RPM" to almost all to my code by the way, Q_(speed, "RPM").

Every thing works well except for "plot_deflected_shape(speed = Q_(speed, "RPM"))".
(FYI, "run_unbalance_response(speed = Q_(speed, "RPM"))" works well.)

Here is the code for "run_unbalance_response()" and "plot_deflected_shape(Q_(speed, "RPM")"

Please correct the code or let me know why it doesn't work.

==================================================
### Unbalance G6.3 on turbine node = 0
n1 = 0
m1 = u1_trans
p1 = 0
###  Unbalance G2.5 on gen' rotor node = 36, out of phase with the turbine unbalance
n2 = 36
m2 = u2_trans
p2 = np.pi

### Unbalance response with speed range in rpm
speed_range=Q_(np.linspace(0, 900, 91), "RPM")
results2 = rotor3.run_unbalance_response([n1, n2], [m1, m2], [p1, p2], speed_range)    # in rpm

### plot the operating Deflected Shape(ODS)
results2.plot_deflected_shape(speed=Q_(514, "RPM"))  <<<<= the error occurs here!
==================================================

**Comment by ViniciusTxc3** [comment, marked as answer]
Hi @KimYoungJoong , thanks for helping in improve ROSS.
Currently, the `.plot_deflected_shape()` function doesn't accept the pint library argument. I'll create an issue with this report and add this possibility.
A current way for your case would be:

```python
results2.plot_deflected_shape(speed=Q_(514, "RPM").to_base_units().m)
```
